helpdesk_mgmt_stage_server_action/tests/test_helpdesk_ticket_stage.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import SavepointCase
+
+
+class HelpdeskTicketStageServerAction(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(HelpdeskTicketStageServerAction, cls).setUpClass()
+        cls.ServerAction = cls.env["ir.actions.server"]
+        cls.HelpdeskTicket = cls.env["helpdesk.ticket"]
+        cls.HelpdeskTicketStage = cls.env["helpdesk.ticket.stage"]
+        cls.field = cls.env["ir.model.fields"]._get(cls.HelpdeskTicket._name, "user_id")
+        cls.server_action_helpdesk_ticket = cls.ServerAction.create(
+            {
+                "name": "Helpdesk Ticket Server Action",
+                "model_id": cls.env.ref("helpdesk_mgmt.model_helpdesk_ticket").id,
+                "state": "object_write",
+                "fields_lines": [
+                    (
+                        0,
+                        0,
+                        {
+                            "col1": cls.field.id,
+                            "evaluation_type": "value",
+                            "value": cls.env.user.id,
+                        },
+                    )
+                ],
+            }
+        )
+        cls.helpdesk_ticket_stage_1 = cls.HelpdeskTicketStage.create(
+            {"name": "Stage 1", "sequence": 1}
+        )
+        cls.helpdesk_ticket_stage_2 = cls.HelpdeskTicketStage.create(
+            {
+                "name": "Stage 2",
+                "action_id": cls.server_action_helpdesk_ticket.id,
+                "sequence": 2,
+            }
+        )
+
+    def test_helpdesk_ticket_create(self):
+        self.helpdesk_ticket_1 = self.HelpdeskTicket.create(
+            {
+                "name": "Helpdesk Ticket 1",
+                "stage_id": self.helpdesk_ticket_stage_2.id,
+                "description": "Helpdesk Ticket Description",
+            }
+        )
+        self.assertEqual(self.helpdesk_ticket_1.user_id, self.env.user)
+
+    def test_helpdesk_ticket_write(self):
+        self.helpdesk_ticket_2 = self.HelpdeskTicket.create(
+            {
+                "name": "Helpdesk Ticket 2",
+                "stage_id": self.helpdesk_ticket_stage_1.id,
+                "description": "Helpdesk Ticket Description",
+            }
+        )
+        self.assertNotEqual(self.helpdesk_ticket_2.user_id, self.env.user)
+        self.helpdesk_ticket_2.write({"stage_id": self.helpdesk_ticket_stage_2.id})
```

Comment:
Please also add a test with writing a ticket without stage_id: the user must not changed.

Do the same with a stage, but no action on it.

Do the same by giving the same stage (with the action) we already have on the ticket.

Please also add a test with several tickets, but a part of them already have the new stage: only tickets on which the stage really changed must have a new user.